src/components/manage/Blocks/Text/Edit.jsx

```diff
@@ -317,32 +287,18 @@ class Edit extends Component {
           }}
         />
         <InlineToolbar />
-        {this.props.selected &&
-          !disableNewBlocks &&
-          !config.blocks.blocksConfig[
-            this.props.data?.['@type'] || 'text'
-          ].blockHasValue(this.props.data) && (
-            <Button
-              basic
-              icon
-              onClick={this.toggleAddNewBlock}
-              className="block-add-button"
-            >
-              <Icon name={addSVG} className="block-add-button" size="24px" />
-            </Button>
-          )}
-        {this.state.addNewBlockOpened && (
-          <BlockChooser
+        {this.props.selected && (
+          <MutateBlockButton
+            data={this.props.data}
+            block={this.props.block}
             onInsertBlock={(id, value) => {
-              this.setState((state) => ({
-                addNewBlockOpened: !state.addNewBlockOpened,
-              }));
               this.props.onSelectBlock(this.props.onInsertBlock(id, value));
             }}
-            currentBlock={this.props.block}
+            onMutateBlock={() => {}}
```

Comment:
why mutating text block is not allowed?

Reply:
There's something fishy in Volto right now:

According to this, the onMutateBlock [parameter is required](https://github.com/plone/volto/blob/master/src/components/manage/BlockChooser/BlockChooser.jsx#L146)

But [it's not passed down](https://github.com/plone/volto/blob/master/src/components/manage/Blocks/Text/Edit.jsx#L335) from the `Text/Edit.jsx` component

I think the wrapping of the component somehow swallows the proptypes warning. I made the params optional, for good measure.